source/event.hpp

```diff
@@ -44,8 +44,8 @@ struct Arrive : public Event {
  * and start processing at the device.
  */
 struct Process : public Event {
-    Process(Device *a_device);
-    Device *node;
+    Process(RountingModule *a_device);
```

Comment:
Should `RoutingModule` be `RoutingDevice`? In this case, 
```
Process(RoutingDevice *a_device);
```
would look better

Reply:
I think it would be better to abstract away here:
```suggestion
    Process(IProcessingDevice *a_device);
```